wave_lang/kernel/ops/wave_ops.py

```diff
@@ -2001,6 +2013,52 @@ def is_contiguous_vec(self) -> bool:
         )
 
 
+@define_op("debug_log_write")
+@dataclass
+class DebugLogBasicWrite(CustomOp):
+    """
+    An op for debugging.
+    Represents a write to an implicit global memory location.
+    The kernel will implicitly have an extra memory input added that will be injected by the Python kernel launcher.
+    The memory can be accessed by passing an an extra keyword to kernel invokation "debug_logs" with an empty dictionary.
+    The dictionary will be mutated to contain the logs.
+
+    This is intended as a primitive for building more convenient debug logging tools.
+    """
+
+    register_: fx.Proxy
+    elements_per_thread: Optional[Any] = None
+    mapping: Optional[IndexMapping] = None
+    mapping_dynamic_vals: tuple[fx.Node, ...] = ()
```

Comment:
This may be a bit too detailed for a debug log, leaking the fact that it is a memory write, but without knowing what memory we are writing into. It can create room for bugs, e.g., by having the user cargo-cult a mapping from another write that is not compatible with the sizes of the memory generated. And, in general, I don't think it's possible to infer the memory size sufficient to cover all addresses after mapping in presence of dynamic values. Do we really need these?

Reply:
Honestly, I'm not 100% sure yet what we need here.  I have written very little wave code, so I'm not yet sure about all the ways these get used.  I left them to be able to manually adjust the writing like you can with normal writes (since they ultimately are just syntax sugar for normal writes), but I'm not yet certain what would be best here.

Reply:
I've removed the extra optional arguments, and they are easy enough to add back if they end up being necessary.